src/app/Common/Caching/Cache.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Common.Caching.Interfaces;
+using Common.Settings;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Common.Caching;
+public class Cache : ICache
+{
+    private readonly IDistributedCache _cache;
+    private readonly ILogger<Cache> _logger;
+    private readonly CacheSettings _cacheSettings;
+
+    public Cache(
+        IDistributedCache cache,
+        ILogger<Cache> logger,
+        IOptions<CacheSettings> cacheSettings)
+    {
+        _cache = cache;
+        _logger = logger;
+        _cacheSettings = cacheSettings.Value;
+    }
+
+    public async Task<T> GetAsync<T>(string key, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            var json = await _cache.GetAsync(key, cancellationToken);
+
+            return json != null
+                ? JsonSerializer.Deserialize<T>(json)
+                : default;
```

Comment:
Maybe it'd be better to restrict T to reference types and return null, if there's a cache miss? Currently, it we try to get an integer with value 0 from the cache, we can't tell for sure, if it's indeed in the cache or not. It can be useful to know this.